### DISCUSSION TITLE
HTTP POST Custom Payload

### DISCUSSION THREAD

**Discussion by 0xyomi** [jertel/elastalert2#1135] [Q&A] [answered]
Hello,
i've made a custom HTTP POST Alerter with this payload as the body :

payload = {
            "case_description": self.create_alert_body(matches)
        }

and it produce this :

id: ucg3oYBaHUVVCPU2Bjp index: falco hostname: falco-kwft5 numhits: 2 nummatches: 2 output: 08:52:00.828425285: Warning Docker or kubernetes client executed in container (user=root userloginuid=-1 k8s.ns=default k8s.pod=system-monitor-deployment-756598dbd-9gfp7 container=35a683b4015d parent=bash cmdline=kubectl get ns pid=4106900 image=madhuakula/k8s-goat-system-monitor:latest) outputfields: { "container.id": "35a683b4015d", "container.image.repository": "madhuakula/k8s-goat-system-monitor", "container.image.tag": "latest", "evt.time": 1679302320828425285, "k8s.ns.name": "default", "k8s.pod.name": "system-monitor-deployment-756598dbd-9gfp7", "proc.cmdline": "kubectl get ns", "proc.pid": 4106900, "proc.pname": "bash", "user.loginuid": -1, "user.name": "root" } priority: Warning rule: The docker client is executed in a container source: syscall tags: [ "T1610", "container", "mitreexecution" ] time: 2023-03-20T08:52:00.828425Z uuid: 8590b196-cf01-4ef1-9488-6ceb9c732c9b

How to make the payload body with some formatting in it?

**Comment by jertel** [comment, marked as answer]
Several other ElastAlert 2 alerters support different types of formatting. Look at those for examples, and find one that formats similar to what you want.

**Reply by 0xyomi** [reply]
how do i get list of the variable i want to include in my alert? like the ID, source, tags, etc?

**Reply by jertel** [reply]
All rule options are available from within the `rule` dict object. The [httppost2](https://github.com/jertel/elastalert2/blob/master/elastalert/alerters/httppost2.py) alerter already included in ElastAlert 2 is a good example to follow.

Why did you choose to write your own alerter instead of using the ones provided? If it's a formatting issue only then you might be better off writing a quick [enhancement](https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html) to format a new field as you would like, and let the existing httppost2 alerter do the rest.

**Reply by 0xyomi** [reply]
i want to try to use DFIR-IRIS as my SOC, so i need to send the alert to the platform. There is no integration to IRIS, so i use the HTTP Post alerter to connect to the IRIS API manually.

**Reply by jertel** [reply]
Understood. But my question is trying to understand what is wrong with the existing HTTP POST alerters? Why weren't you able to use them?

**Comment by baphael** [comment]


**Reply by jertel** [reply]
Please start a new thread per #11.
